src/x-platform/index.js

```diff
@@ -1,26 +1,107 @@
+import 'html5-desktop-notifications'
+import {notificationsConfig} from './constants'
+import noop from 'lodash/utility/noop'
+
+
+// Electron brings own require to window
+let electron
+let remote
+if (window.require) {
+  electron = require('electron')
+  remote = require('remote')
+}
+
+const macGap = window.MacGap
```

Comment:
as an idea: create separate files for macgap, electron and the web which implement the same functions. Then decide in index which one to export. This way we can easily remove any of it later, without to grow the "ifs"